packages/lavapack/src/runtime.js

```diff
@@ -11147,24 +11147,19 @@ module.exports = {
     const scuttleGlobalThisDefaults = {
       enabled: true,
       exceptions: [],
-      recursive: false,
+      scuttler: '',
     }
-    const {
-      enabled: scuttleGlobalThisEnabled,
-      exceptions: scuttleGlobalThisExceptions,
-      recursive: scuttleGlobalThisRecursive,
-    } = scuttleGlobalThis === true ? scuttleGlobalThisDefaults : scuttleGlobalThis
-
-    let snow = (cb, win) => cb(win)
-    if (scuttleGlobalThisRecursive) {
-      if (!globalRef.SNOW) {
+
+    const scuttleOpts = scuttleGlobalThis === true ? scuttleGlobalThisDefaults : scuttleGlobalThis
+    scuttleOpts.scuttlerFunc = (globalRef, scuttle) => scuttle(globalRef)
+    if (scuttleOpts.scuttler) {
+      if (!globalRef[scuttleOpts.scuttler]) {
         throw new Error(
-          'LavaMoat - scuttleGlobalThis is configured to be enabled recursively. ' +
-          'For that, Snow-JS must be included before LavaMoat executes. ' +
+          `LavaMoat - 'scuttler' function "${scuttleOpts.scuttler}" expected on globalRef.` +
           'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/462.',
         )
       }
-      snow = globalRef.SNOW
+      scuttleOpts.scuttlerFunc = globalRef[scuttleOpts.scuttler]
     }
```

Comment:
maybe should create a configuration obj that holds the scuttlerFunc instead of mutating the options obj

Reply:
bb15afce1